demos/react-jsnation-workshop/src/components/steps/Step4.tsx

```diff
@@ -0,0 +1,129 @@
+import { ReactiveMap } from "@starbeam/collections";
+import { useReactive, useResource } from "@starbeam/react";
+import { Resource, type ResourceBlueprint } from "@starbeam/universal";
+import { useState } from "react";
+
+import type { ApiUser } from "../shared/api.js";
+import type { Async } from "../shared/async.js";
+import { RemoteData } from "../shared/remote-data.js";
+import { Field, Jsonify } from "../shared/shared.js";
+
+export default function App(): JSX.Element {
+  const [currentId, setCurrentId] = useState("1");
+  const users = useResource(Users);
+
+  return useReactive(() => {
+    const user = users.get(Number(currentId));
+
+    return (
+      <section className="user-grid">
+        <div className="user-list">
+          <Field
+            type="number"
+            label="User ID"
+            min={1}
+            max={10}
+            onUpdate={setCurrentId}
+            value={currentId}
+          />
+          <div className="user-list">
+            {users.all().map(([id, user]) => (
```

Comment:
does this stay stable when properties of a user change?

last I played around with type of UI and starbeam, changing properties in a user ended up dirtying the parent component.

Looks like `user` is an object that would have delayed evaluation and UserItem has useReactive, so I think the I ran in to would not be present here.